docs/DeveloperGuide.adoc

```diff
@@ -767,16 +821,255 @@ Use case ends.
 +
 Use case resumes at step 2.
 
-{More to be added}
+[discrete]
+=== Use case: Edit person
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to edit a specific person in the list
+4.  AddressBook update the person’s contact details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+* 3b. New contact’s details provided by user are invalid.
++
+[none]
+** 3b1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Renaming of an existing tag
+
+**MSS**
+
+1.  User requests to list all existing tags
+2.  AddressBook shows a list of existing tags
+3.  User requests to rename an existing tag
+4.  AddressBook prompts the confirmation of request
+5.  User confirms the request
+6.  AddressBook renames the existing tag <br>
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. Tag not found.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 5a. User says no
```

Comment:
Here should be `4a.` Check others numbering too.

In use case extensions, the numbering is use to describe scenarios that can happen, preventing the user from going to the next step. 
So in this case, User cancels the change prevent the program from going to step 5.

4. AddressBook confirms the change with the user
`4a.` User cancels change
5. User confirms change